### ISSUE TITLE
[Request] Automatic Decky & Plugin  Updates on Startup

### ISSUE THREAD

**Issue by CreeperofDoom** [SteamDeckHomebrew/decky-loader#658]
### Please confirm

- [X] I have searched existing issues
- [X] This issue is not a duplicate of an existing one
- [X] This is not a request for a plugin

### Feature Request Description

Adding Automatic Updates: Making it where a Decky Loader Update will automatically download and startup and apply on Reboot.

Also being able to mark Decky Plugins to Update Automatically in the Background when Available (EX: CSS Loader)

Maybe Tag it as Experimental

I primarily want this for ease of use. I also installed Decky onto my Friend's Steam Deck's and it keeps breaking because they wont know how to update it

### Further Description

Ease of Use, starting my Steam Deck and being able to get what i want to be doing and not be annoyed by Updates would be Great. Just like how the Steam Deck's own Updates are automatic.

**Comment by AkazaRenn** [comment]
How about we patch `SteamClient.Updates.ApplyUpdates` to also update Decky Loader, in this way when Steam Updates we make sure Decky is also updated to avoid old version Decky crashing new version Steam.

`SteamClient.Updates.ApplyUpdates("CAE=")` means client update. The string `"CAE=" come from 
```ts
btoa('\b' + String.fromCharCode(updateType)); // updateType == 1 for client update
```
I assume the updateType value comes from https://github.com/SteamDatabase/Protobufs/blob/master/steam/enums.proto#L279-L288, since for OS update the base64String will be "CAI=" which is converted when `updateType` is `2`.

**Comment by AAGaming00** [comment]
good idea though I'd prefer to show a prompt and we wouldn't do it by patching SteamClient since that tends to jank out a tad

**Comment by AkazaRenn** [comment]
> good idea though I'd prefer to show a prompt and we wouldn't do it by patching SteamClient since that tends to jank out a tad

Then it might be easier to do it on start up, maybe at the crash page add new button for updating Decky and a button to switch channel

**Comment by AAGaming00** [comment]
> > good idea though I'd prefer to show a prompt and we wouldn't do it by patching SteamClient since that tends to jank out a tad
> 
> Then it might be easier to do it on start up, maybe at the crash page add a new button for updating Decky and a button to switch channel

ideally both
